guides/plugins/plugins/framework/event/finding-events.md

```diff
@@ -0,0 +1,184 @@
+You can use them by following this pattern: `entity_name.event`.
+For products, this could be e.g. `product.written` or `product.deleted`. For your custom entity, this then would be 
+`custom_entity.written` or `custom_entity.deleted`.
+
+However, some default Shopware entities come with special "Event classes", which are basically a class, which contains all
+possible kinds of events as constants.
+Have a look at the [product event class](https://github.com/shopware/platform/blob/v6.4.0.0/src/Core/Content/Product/ProductEvents.php) for example.
+This way you can also find out about all the possible DAL events available in Shopware.
+
+Finding those "event classes" can be done by searching for the term `@Event` in your project.
+
+You can use those events in a [subscriber](../../plugin-fundamentals/listening-to-events.md) like the following:
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        ProductEvents::PRODUCT_LOADED_EVENT => 'onProductsLoaded',
+        'custom_entity.written' => 'onCustomEntityWritten'
+    ];
+}
+```
+
+As you can see, you can either use the event class constants, if available, or the string itself.
+
+You'll then have access to several event specific information, e.g. your listener method will have access to an [EntityWrittenEvent](https://github.com/shopware/platform/blob/v6.4.0.0/src/Core/Framework/DataAbstractionLayer/Event/EntityWrittenEvent.php)
+instance when subscribing to the `written` event.
+
+```php
+public function onCustomEntityWritten(EntityWrittenEvent $event): void
+{
+}
+```
+
+You can find all of those DAL event classes [here](https://github.com/shopware/platform/tree/v6.4.0.0/src/Core/Framework/DataAbstractionLayer/Event).
+
+## Other PHP events
+
+If the [DAL events](#DAL events) do not suffice for your use case, there are a few other events built into Shopware.
+These are not auto-generated events, but rather events we built in with purpose.
+
+There are multiple ways to find them:
+- By actually looking at the code, that you want to extend
+- By specifically searching for them
+- By having a look at the service definition of a given class
+
+### Looking at the code
+
+You will most likely look into our Core code quite a lot, while trying to understand what's happening and why things are happening.
+On your journey looking through the code, you may stumple upon code looking like this:
+
+```php
+$someEvent = new SomeEvent($parameters, $moreParameters);
+$this->eventDispatcher->dispatch($someEvent, $someEvent->getName());
+```
+
+This is an event that's being fired manually, which you can react upon.
+Make sure to always have a look at the event class itself in order to find out which information it contains.
+
+When subscribing to those events, your event listener method will have access to the previously created event instance. 
+
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        SomeEvent::class => 'registeringToSomeEvent'
```

Comment:
It probably will not look like that. When you dispatch an event by its name it is rarely dispatched by its class name. This is a pretty common case for validation or entity events as their name is dynamic

Reply:
Not too sure what you mean here, to be honest. :-(

Reply:
This relates to the line `$this->dispatcher->dispatch($event, $event->getName());`. There is only one reason to add the second parameter for the name: when the name is not the class but a different one. And now you explain to subscribe to the class name. But `getName()` probably won't return the class name.

Reply:
Good hint in the sample code. Works out for me.